src/views/Timesheets/components/SavedShiftsList/index.js

```diff
@@ -0,0 +1,306 @@
+                    Cancel
+                  </Button>
+                </Grid>
+                <Grid item xs={6} sm={6} md={6} lg={6}>
+                  <Button
+                    variant="contained"
+                    onClick={() => {
+                      this.submitShiftsToSupervisor(this.props.shifts, this.state.selectedSupervisor.id)
+                    }}
+                    color="primary">
+                    Submit
+                  </Button>
+                </Grid>
+              </Grid>
+            </DialogContent>
+          </Dialog>
+        </Grid>
+      </Grid>
+    );
+
+    let header = (
+      <Grid item xs={12} style={styles.headerStyle}>
+        <div>
+          <Grid container direction="row">
+            <Grid item xs={3}>
+              <Typography variant="body2" style={styles.headerItem}>
+                JOB
+              </Typography>
+            </Grid>
+            <Grid item xs={2}>
+              <Typography variant="body2" style={styles.headerItem}>
+                TIME IN
+              </Typography>
+            </Grid>
+            <Grid item xs={2}>
+              <Typography variant="body2" style={styles.headerItem}>
+                TIME OUT
+              </Typography>
+            </Grid>
+            <Grid item xs={2}>
+              <Typography variant="body2" style={styles.headerItem}>
+                RATE
+              </Typography>
+            </Grid>
+            <Grid item xs={2}>
+              <Typography variant="body2" style={styles.headerItem}>
+                HOURS WORKED
+              </Typography>
+            </Grid>
+          </Grid>
+        </div>
+      </Grid>
+    );
+
+    let shiftsList = null;
+    shiftsList = this.props.shifts.map((shift, index) => (
+      <ShiftItem deleteShift={this.props.deleteShift} value={shift} key={index} />
+    ));
+
+    const supervisorDropdown = (
+      <FormControl
+        style={{
+          width: 252,
+        }}
+      >
+        <InputLabel>Submit To</InputLabel>
+        <Select
+          value={this.state.selectedSupervisor}
+          onChange={e => {
+            this.setState({
+              selectedSupervisor: e.target.value,
+            });
+          }}
+          input={<Input id="supervisor" />}
+        >
+          <MenuItem label="direct supervisor" value={this.state.directSupervisor}>
+            <Typography>
+              {this.state.directSupervisor !== null && this.state.directSupervisor.name}
+            </Typography>
+          </MenuItem>
+          <MenuItem label="reporting supervisor" value={this.state.reportingSupervisor}>
+            <Typography>
+              {this.state.reportingSupervisor !== null && this.state.reportingSupervisor.name}
+            </Typography>
+          </MenuItem>
+        </Select>
+      </FormControl>
+    );
+
+    let content = <></>;
+    if (this.props.shifts.length === null) {
```

Comment:
Will shifts.length be zero if the user has no shift history, or will the Loader show for that case also?

Reply:
If a saved shifts list is empty, it will not show the card